src/bt-core-io.c

```diff
@@ -507,31 +506,65 @@ alarm_event(int fd, uint32_t events, void* data)
   return res;
 }
 
-static bool
-set_wake_alarm_cb(uint64_t delay_millis, bool should_wake, alarm_cb cb,
-                  void* data)
+struct set_wake_alarm_param {
+  uint64_t delay_millis;
+  bool should_wake;
+  alarm_cb cb;
+  void* data;
+};
+
+static enum ioresult
+set_wake_alarm_task_cb(void* data)
 {
+  struct set_wake_alarm_param* param = data;
   struct alarm_state* alarm_state;
   int clockid;
+  assert(param);
 
-  errno = 0;
   alarm_state = malloc(sizeof(*alarm_state));
-  if (errno) {
+  if (!alarm_state) {
```

Comment:
Good :) Unfortuantely we found a platform where errno gets set by successful mallocs. So testing for !NULL is the better option.